File: Wizzi/ClientApp/src/dtos/SolicitudesCitasMedicas/verSolicitudCitaMedicaDto.ts
import { VerCitaMedicaDto } from "../CitasMedicas";
import { VerPaisDto, VerProvinciaDto, VerCantonDto, VerParroquiaDto } from "../Localizaciones";
import { VerRelacionRepresentantePaciente } from "../RelacionesRepresentantes";
import { SubCampaniasDto } from "../Subcampanias";
import { VerSucursalDto } from "../Sucursales";
import { _BaseDto } from "./_BaseDto";

export interface VerSolicitudCitaMedicaDto extends _BaseDto {
    subcampaniaOrigen: SubCampaniasDto;
    cita: VerCitaMedicaDto;
    localizacion: LocalizacionDto;
    relacionRepresentantePaciente: VerRelacionRepresentantePaciente;
    sucursal: VerSucursalDto;
    fechaRegistro: Date;
    cantidadAgendas: number;
    cantidadLlamadasUltimoProceso: number;
    cantidadMovimientos: number;
}

export interface LocalizacionDto {
    pais: VerPaisDto;
    provincia: VerProvinciaDto;
    canton: VerCantonDto;
    parroquia: VerParroquiaDto;
}

File: Wizzi/Dtos/SolicitudesCitasMedicas/VerSolicitudCitaMedicaDto.cs
using System;
using Wizzi.Dtos.CitasMedicas;
using Wizzi.Dtos.Localizaciones;
using Wizzi.Dtos.RelacionesRepresentantes;
using Wizzi.Dtos.Subcampanias;
using Wizzi.Dtos.Sucursales;

namespace Wizzi.Dtos.SolicitudesCitasMedicas
{
    public class VerSolicitudCitaMedicaDto : _BaseDto
    {
        public SubCampaniasDto subcampaniaOrigen { get; set; }
        public VerCitaMedicaDto cita { get; set; }
        public LocalizacionDto localizacion { get; set; }
        public VerRelacionRepresentantePaciente relacionRepresentantePaciente { get; set; }
        public VerSucursalDto sucursal { get; set; }
        public DateTime fechaRegistro { get; set; }
        public int cantidadAgendas { get; set; }
        public int cantidadLlamadasUltimoProceso { get; set; }
        public int cantidadMovimientos { get; set; }
    }

    public class LocalizacionDto
    {
        public VerPaisDto Pais { get; set; }
        public VerProvinciaDto Provincia { get; set; }
        public VerCantonDto Canton { get; set; }
        public VerParroquiaDto Parroquia { get; set; }
    }

    //public class VerSucursalDto
    //{
    //    public string Codigo { get; set; }
    //    public string Nombre { get; set; }
    //}

}
